src/main/java/org/kumoricon/registration/admin/roles/UserRightsUpdateInterceptor.java

```diff
@@ -0,0 +1,63 @@
+package org.kumoricon.registration.admin.roles;
+
+import org.kumoricon.registration.model.role.Right;
+import org.kumoricon.registration.model.role.RightRepository;
+import org.kumoricon.registration.model.user.User;
+import org.kumoricon.registration.model.user.UserRepository;
+import org.springframework.security.authentication.AbstractAuthenticationToken;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+
+/**
+ * Interceptor to update user rights with saved rights from database. This is to allow user rights to be updated
+ * without requiring logging in/out
+ */
+@Component
+public class UserRightsUpdateInterceptor implements HandlerInterceptor {
+
+
+    private final RightRepository rightRepository;
+    private final UserRepository userRepository;
+
+    public UserRightsUpdateInterceptor(RightRepository rightRepository, UserRepository userRepository) {
+        this.rightRepository = rightRepository;
+        this.userRepository = userRepository;
+    }
+
+
+    /**
+     * Updates user authorities with the newest authorities loaded from database
+     * TODO: checking the database for every intercepted request is expensive, maybe cache to lower db calls
+     */
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+
+        final Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        final Set<GrantedAuthority> authorities = new HashSet<>();
+        Object principal = auth.getPrincipal();
+
+        if (principal instanceof User user) {
+            // Get User and User's Rights from database
+            User userFromDatabase = userRepository.findOneByUsernameIgnoreCase(user.getUsername());
```

Comment:
im very dumb i just realized this isnt needed, we can use user principal id directly